aten/src/ATen/native/cuda/TriangularOps.cu

```diff
@@ -19,33 +19,53 @@
 
 #include <ATen/cuda/CUDAApplyUtils.cuh>
 
+#define BOOL_SWITCH(COND, CONST_NAME, ...)      \
```

Comment:
I do notice we already have a definition of this in aten/native/transformers/cuda/flash_attn/static_switch.h, I wonder if this could be abstracted to a higher level, that is a common ancestor of the 2. This can come in a later PR. cc @drisspg

Reply:
yeah, somewhere in c10 util would likely be a good location

Reply:
@janeyx99 There is also a similar macro called `DISPATCH_BOOL` in mem_eff_attention/gemm_kernel_utils.h:

https://github.com/pytorch/pytorch/blob/fee96adde766c401aaaeb3fda3e5240e084885d7/aten/src/ATen/native/transformers/cuda/mem_eff_attention/gemm_kernel_utils.h#L31-L40

cc @drisspg